### DISCUSSION TITLE
Requirement of More CircuitPython examples on IDM2040-7A Bridgetek Display Modules Display module

### DISCUSSION THREAD

**Discussion by Alpha-Amol** [Bridgetek/pico-brteve#26] [Q&A]
### Have proper documentation on Circuitpython library (pico-brteve) for IDM2040-7A Bridgetek Display Modules Display module 7 capacitive touch panel with BT817Q eve chip? 

__**Have Simple examples Like:**__ 

1) How to load a image using EVE Asset Builder into the External Flash connected to BT817 and displaying the same image using circuitpython (pico-brteve library)?

2) How to load multiple images using EVE Asset Builder into the External Flash connected to BT817 and displaying the images one by one using circuitpython (pico-brteve library)? 

3) How to load a font of a particular size using EVE Asset Builder into the External Flash connected to BT817 and displaying the text in that font using circuitpython (pico-brteve library)? 

4) How to load multiple fonts of multiple sizes using EVE Asset Builder into the External Flash connected to BT817 and displaying texts in those fonts with different sizes and  using circuitpython (pico-brteve library)? 

5) How to load a gif animation using EVE Asset Builder into the External Flash connected to BT817 and displaying the same gif using circuitpython (pico-brteve library)? 

6) How to load multiple gif animations using EVE Asset Builder into the External Flash connected to BT817 and displaying gifs one by one using circuitpython (pico-brteve library)? 

7) How to access touch features using buttons and keys with simple examples? Like how to use touch tag registers with buttons, keys, or other widgets with touch features? 

8) How to slide the multiple images stored in flash memory one by one?

Waiting for your reply!

**Comment by brtchip-tuannguyen** [comment]
Hi @Alpha-Amol ,


q1. How to load a image using EVE Asset Builder into the External Flash connected to BT817 and displaying the same image using circuitpython (pico-brteve library)?

   - Document about generate image by EAB: [Section E. Image Utilities](https://brtchip.com/wp-content/uploads/2023/03/BRT_AN_054-EVE-Asset-Builder-2.8-User-Guide.pdf)
- There's no direct example for this puspose, but we can see a similar reference here: [alarm_clock/helper.py#209](https://github.com/Bridgetek/pico-brteve/blob/5faba01334f5bfc0d8ae0248a481c69a8c2bdbc6/circuitPython/examples/alarm_clock/alarm_clock/helper.py#L209)

q2. How to load multiple images using EVE Asset Builder into the External Flash connected to BT817 and displaying the images one by one using circuitpython (pico-brteve library)?

   1. User needs to gather images file into a flash binary file (example: BT81X_Flash.bin), use EAB.
   2. User write the flash binary file (BT81X_Flash.bin) into External Flash connected to BT817, use EAB.
   3. Use CircuitPython code to display images in the External Flash connected, refer: [page_charge.py#L143](https://github.com/Bridgetek/pico-brteve/blob/5faba01334f5bfc0d8ae0248a481c69a8c2bdbc6/circuitPython/examples/ev-charge-point/ev-charge-point/page_charge.py#L143).
   
   
q3. q4. q5. q6:
- There's no existing example for this purpose in CircuitPython, we can refer C code example in EveApps.


q7. How to access touch features using buttons and keys with simple examples? Like how to use touch tag registers with buttons, keys, or other widgets with touch features?

 - Can refer ev-charge-point/gesture.py 


q8. How to slide the multiple images stored in flash memory one by one?

  - Currently no existing example for this purpose in CircuitPython, but there's an example to slide JPG image: image-slide-viewer   

- To slide multiple images stored in flash memory:
        1. The JPG can be transfered into ASTC, gather into .bin file and written into flash memory at first.
        2. To slide images, draw the ASTC images with vertex(x,y), change the (x,y) position to make the slide animation.
